azurelinuxagent/common/cgroup.py

```
@@ -86,23 +86,28 @@ def _get_parameters(self, parameter_name, first_line_only=False):
         except IndexError:
             parameter_filename = self._get_cgroup_file(parameter_name)
             logger.error("File {0} is empty but should not be".format(parameter_filename))
+            raise CGroupsException("File {0} is empty but should not be".format(parameter_filename))
         except CGroupsException:
-            return None
+            raise
         except Exception as e:
             parameter_filename = self._get_cgroup_file(parameter_name)
             logger.error("Exception while attempting to read {0}: {1}".format(parameter_filename, ustr(e)))
+            raise CGroupsException("Exception while attempting to read {0}: {1}".format(parameter_filename, ustr(e)))
         return result
 
     def collect(self):
         raise NotImplementedError()
 
     def is_active(self):
-        tasks = self._get_parameters("tasks")
-        if tasks:
-            return len(tasks) != 0
-        else:
+        try:
+            tasks = self._get_parameters("tasks")
+            if tasks:
+                return len(tasks) != 0
+        except CGroupsException:
```

this will mask exceptions; should probably log the error